src/edit.js

```diff
@@ -2,43 +2,142 @@
  * External dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { useBlockProps, InspectorControls } from '@wordpress/block-editor';
-import { PanelBody, TextControl } from '@wordpress/components';
+import {
+	useBlockProps,
+	InspectorControls,
+	PlainText,
+} from '@wordpress/block-editor';
+import { PanelBody, TextControl, Notice } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
-import Block from './block';
+import ProgressBar from './components/progress-bar';
+import ViewSwitcher from './components/view-switcher';
+import BlockErrorBoundary from './components/block-error-boundary';
+import './style.scss';
 
 export default function Edit( { attributes, setAttributes } ) {
-	const { freeShippingFrom } = attributes;
+	const {
+		labelInsufficientTotals,
+		labelSufficientTotals,
+		freeShippingFrom,
+	} = attributes;
 
 	return (
 		<div { ...useBlockProps() }>
-			<InspectorControls>
-				<PanelBody
-					title={ __( 'Settings', 'free-shipping-progress-bar' ) }
-					initialOpen={ true }
-				>
-					<TextControl
-						label={ __(
-							'Free shipping from',
+			<ViewSwitcher
+				label={ __( 'Edit', 'free-shipping-progress-bar' ) }
+				views={ [
+					{
+						value: 'insufficient',
+						name: __(
+							'Insufficient cart totals',
+							'free-shipping-progress-bar'
+						),
+					},
+					{
+						value: 'sufficient',
+						name: __(
+							'Sufficient cart totals',
+							'free-shipping-progress-bar'
+						),
+					},
+				] }
+				defaultView={ 'insufficient' }
+				render={ ( currentView ) => (
+					<BlockErrorBoundary
+						header={ __(
+							'Progress Bar Block Error',
+							'free-shipping-progress-bar'
+						) }
+						text={ __(
+							'There was an error whilst editing the progress bar block. If this problem continues, try re-creating the block.',
 							'free-shipping-progress-bar'
 						) }
-						help={ __(
-							'Provide the value in your store currency.',
+						showErrorMessage={ true }
+						errorMessagePrefix={ __(
+							'Error message:',
 							'free-shipping-progress-bar'
 						) }
-						value={ freeShippingFrom }
-						onChange={ ( value ) =>
-							setAttributes( {
-								freeShippingFrom: value,
-							} )
-						}
-					/>
-				</PanelBody>
-			</InspectorControls>
-			<Block { ...attributes } />
+					>
+						{ currentView === 'insufficient' && (
+							<>
+								<InspectorControls>
```

Comment:
Navigating this code is extremely hard now with this level of indention. Can you abstract `InspectorControls` into its own component above and call it here? You can check our Cart and Checkout blocks for examples.

Reply:
As requested, I moved `<InspectorControls>` into a separate component.